### PR TITLE
Update javascript-9-async.md

### LINKED ISSUE #1491: Improve promise example in JS -9 Async Lesson
https://github.com/Techtonica/curriculum/blob/main/javascript/javascript-9-async.md

JS Code snippet: https://github.com/Techtonica/curriculum/blame/main/javascript/javascript-9-async.md#L94-L124

I found the second example after 5. in guided practice a bit confusing. It was unclear to me for a while that there are in fact 2 promises in this example . I think it would be helpful to include comments in the example code and a description before or after the example going through what happened. It might also be helpful to see a video walk-though/voice over explaining the code line by line!

### PR DESCRIPTION
**Pull request by Yosolita1978**
Update the current code example, and add a new code example of working with promises. Adding also a video with an explanation line by line

Fixes #1491

### CODE REVIEW

File: javascript/javascript-9-async.md [hunk 1]
```diff
@@ -90,43 +90,81 @@ There's got to be an easier way to write things that depend on each other, right
...
+    })
+    //The catch method allows you to manage errors inside Promise 
+    .catch(() => {
+      console.log('No internet connection found. App is running in offline mode');
+    });
+}
 ```
+- Watch [this video with the explanation of the code example](https://www.youtube.com/watch?v=_4Y7ly8k0j4&ab_channel=CrissRodriguez). 
 
-### Independent Practice
```
Comment (alodahl) [inline javascript/javascript-9-async.md:127]: Why remove the markdown header syntax?

File: javascript/javascript-9-async.md [hunk 2]
```diff
@@ -90,43 +90,81 @@ There's got to be an easier way to write things that depend on each other, right
...
+        registerValid(Url);
+      }
+    })
+    //The catch method allows you to manage errors inside Promise 
+    .catch(() => {
+      console.log('No internet connection found. App is running in offline mode');
+    });
+}
 ```
+- Watch [this video with the explanation of the code example](https://www.youtube.com/watch?v=_4Y7ly8k0j4&ab_channel=CrissRodriguez).
```
Comment (alodahl) [inline javascript/javascript-9-async.md:163]: 😍

Comment (alodahl) [inline javascript/javascript-9-async.md:163]:
```suggestion
- Watch [this 4 minute video with the explanation of the code example](https://www.youtube.com/watch?v=_4Y7ly8k0j4&ab_channel=CrissRodriguez). 
```

Comment (Yosolita1978) [reply javascript/javascript-9-async.md:163]: Ohhh you're right.

File: javascript/javascript-9-async.md (hunk 1 above)
Comment (aedward8) [reply javascript/javascript-9-async.md:127]: @alodahl  I did not make this current pull request. Did you want me to review it as a current participant to see what I think ? Or did you accidentally tag me in this review thinking that I made the request?

Comment (alodahl) [reply javascript/javascript-9-async.md:127]: Yes, please review! since you’re the one who asked for the change, I want to make sure it’s what you needed.

**Review by alodahl — APPROVED** [review]

**Pull request merged**